client/recordUtil.js

```diff
@@ -285,6 +285,16 @@ module.exports.resolveRecords = (recordsAndExistingObjects) => {
       normalizeBookmark(item[1])
     }
   })
+
+  // Remove obvious conflicting records
+  for (var i = recordsAndExistingObjects.length - 1; i >= 0; --i) {
+    if (recordsAndExistingObjects[i] && recordsAndExistingObjects[i][0]) {
+      if (recordsAndExistingObjects[i][0].action === proto.actions.CREATE && recordsAndExistingObjects[i][1]) {
```

Comment:
what is the obvious conflict? If this is to implement ignore resolving records from self, then we should at least check device id instead of blindly stripping it

Reply:
obvious conflict is:
`server_record.action == CREATE`
and `local_record != null`

There is no point to keep such resolve pair:
1) If this pair is the only pair for given object_id, it will be ignored
2) If there are some other pair for given object_id
2.1) in the case of other record  UPDATE, will be created CREATE record with merged data, and then it will be thrown out, because `local_record != null`
2.2) in the case of other record DELETE , ignoring first [CREATE , object] will not harm

Reply:
can we add 2.2 to test case?
And 2.1 is not right, UPDATE will only be converted into CREATE when there is no local record